Library/Homebrew/downloadable.rb

```diff
@@ -121,9 +121,9 @@ def determine_url
     @url
   end
 
-  sig { overridable.returns(T::Array[String]) }
+  sig { overridable.returns(T::Array[URL]) }
   def determine_url_mirrors
-    [determine_url.to_s, *mirrors].uniq
+    [*([determine_url] unless determine_url.nil?), *mirrors].uniq
```

Comment:
When would URL being nil here be valid?

The caller of `determine_url_mirrors` explicitly checks if the first part is nil, and if not, errors.

It sounds like this should only happen when you have a list of mirrors but no primary URL, which doesn't sound like a valid formula or cask.